src/structure.rs

```diff
@@ -128,3 +268,790 @@ impl<'a> Resources<'a> {
 }
 
 deref!('a, Resources<'a> => Dict<'a>, dict);
+
+/// How the viewer should lay out the pages in the document.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum PageLayout {
+    /// Only a single page at a time.
+    SinglePage,
+    /// A single, continously scrolling column of pages.
+    OneColumn,
+    /// Two continously scrolling columns of pages, laid out with odd-numbered
+    /// pages on the left.
+    TwoColumnLeft,
+    /// Two continously scrolling columns of pages, laid out with odd-numbered
+    /// pages on the right (like in a left-bound book).
+    TwoColumnRight,
+    /// Only two pages are visible at a time, laid out with odd-numbered pages
+    /// on the left. (1.5+)
+    TwoPageLeft,
+    /// Only two pages are visible at a time, laid out with odd-numbered pages
+    /// on the right (like in a left-bound book). (1.5+)
+    TwoPageRight,
+}
+
+impl PageLayout {
+    fn to_name(self) -> Name<'static> {
+        match self {
+            Self::SinglePage => Name(b"SinglePage"),
+            Self::OneColumn => Name(b"OneColumn"),
+            Self::TwoColumnLeft => Name(b"TwoColumnLeft"),
+            Self::TwoColumnRight => Name(b"TwoColumnRight"),
+            Self::TwoPageLeft => Name(b"TwoPageLeft"),
+            Self::TwoPageRight => Name(b"TwoPageRight"),
+        }
+    }
+}
+
+/// Elements of the viewer chrome that should be visible when opening the
+/// document.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum PageMode {
+    /// Neither the document outline panel nor a panel with page preview images
+    /// are visible.
+    UseNone,
+    /// The document outline panel is visible.
+    UseOutlines,
+    /// A panel with page preview images is visible.
+    UseThumbs,
+    /// Show the document page in full screen mode, with no chrome.
+    FullScreen,
+}
+
+impl PageMode {
+    fn to_name(self) -> Name<'static> {
+        match self {
+            Self::UseNone => Name(b"UseNone"),
+            Self::UseOutlines => Name(b"UseOutlines"),
+            Self::UseThumbs => Name(b"UseThumbs"),
+            Self::FullScreen => Name(b"FullScreen"),
+        }
+    }
+}
+
+/// Predominant reading order of text. Used to aid the viewer with the spacial
+/// ordering in which to display pages.
```

Comment:
```suggestion
/// Predominant reading order of text.
///
/// Used to aid the viewer with the spacial ordering in which to display pages.
```
looks better in docs